### PR TITLE
Fix react-redux 6.0.0 new way of passing context

### PR DESCRIPTION
**Pull request by GuillaumeCisco**
Fix https://github.com/faceyspacey/redux-first-router-link/issues/101

### CODE REVIEW

File: src/Link.js [hunk 1]
```diff
@@ -98,11 +90,10 @@ export const Link = (
...
 
-Link.contextTypes = {
-  store: PropTypes.object.isRequired
-}
-
-const connector: Connector<OwnProps, Props> = connect()
+const mapState = state => ({
+  routesMap: selectLocationState(state).routesMap
+})
+const connector: Connector<OwnProps, Props> = connect(mapState)
```
Comment (cdoublev) [inline src/Link.js]: Shouldn't `dispatch` be mapped to props here?

File: src/NavLink.js [hunk 2]
```diff
@@ -37,39 +36,31 @@ type OwnProps = {
 
 type Props = {
   dispatch: Function,
```
Comment (cdoublev) [inline src/NavLink.js]: Line 39 could be removed, I think.

File: src/Link.js (hunk 1 above)
Comment (GuillaumeCisco) [reply src/Link.js]: Hum, what do you have in mind?
Do you see a case dispatch is needed?

Comment (cdoublev) [reply src/Link.js]: It's already used and in a Redux context, dispatching action is needed, I guess. :)

Comment (GuillaumeCisco) [reply src/Link.js]: `dispatch` was not present before, is this necessary?
I wonder

Comment (GuillaumeCisco) [reply src/Link.js]: Seems not mandatory: https://stackoverflow.com/questions/34458261/how-to-get-simple-dispatch-from-this-props-using-connect-w-redux

**Comment by ScriptedAlchemy** [comment]
If still relevant @GuillaumeCisco @cdoublev 

Please merge with master and resolve conflicts, then ill merge it if the community (you guys) say yes

**Comment by ScriptedAlchemy** [comment]
Ill merge this if we can resolve the conflicts. Otherwise give me like 4 hours and ill do so on my machine. Again I’m really sorry i got PR's mixed up

File: src/Link.js (hunk 1 above)
Comment (cdoublev) [reply src/Link.js]: And this is [officially supported](https://react-redux.js.org/api/connect#mapdispatchtoprops-object-dispatch-ownprops-object), ok. 👍

**Comment by GuillaumeCisco** [comment]
Thanks @cdoublev and @ScriptedAlchemy.
As we are in Christmas days, I cannot access a computer right now. I will in 2 days.
If someone can make a PR from this PR, it will be great!
We could merge it from GitHub gui :)

**Comment by ScriptedAlchemy** [comment]
Sure thing I'll update your pr tomorrow and merge. Tomorrow is my last day before family vacation to the mountains. No idea what wireless service is there. So I'll make sure this is merge tomorrow. Merry Christmas or happy holidays!!!

**Comment by ScriptedAlchemy** [comment]
Checking in here, we all good?

**Comment by cdoublev** [comment]
Yes, thank you both. I just updated it with react-redux: links are working correctly. 👍

**Comment by ScriptedAlchemy** [comment]
Thanks guys. I really appreciatate it!

**Pull request merged**